perceval/backends/core/github.py

```diff
@@ -704,19 +719,76 @@ def fetch_items(self, path, payload):
                 items = response.text
                 logger.debug("Page: %i/%i" % (page, last_page))
 
+    def _choose_best_api_token(self):
+        """Check all API tokles defined and choose one with most remaining API points"""
+
+        url = urijoin(self.base_url, "rate_limit")
+        max_idx = 0
+        # If multiple tokens given, choose best
+        if len(self.token) > 1:
+            hdrs = [0] * len(self.token)
+            # Turn off archiving when checking rates, because that would cause
+            # archive key conflict (the same URLs giving different responses)
+            arch = self.archive
+            self.archive = False
+            for idx, tok in enumerate(self.token):
+                self.session.headers.update({'Authorization': 'token ' + tok})
+                try:
+                    hdrs[idx] = super().fetch(url).headers
+                except requests.exceptions.HTTPError as error:
+                    if error.response.status_code == 404:
+                        logger.warning("Rate limit not initialized: %s", error)
+                    else:
+                        # Restore archiving to whatever state it was
+                        self.archive = arch
+                        raise error
+            # Restore archiving to whatever state it was
+            self.archive = arch
+            rls = [0] * len(self.token)
+            for idx, hdr in enumerate(hdrs):
+                if self.rate_limit_header in hdr:
+                    rls[idx] = int(hdr[self.rate_limit_header])
+                else:
+                    rls[idx] = 0
+            max_idx = rls.index(max(rls))
+            self.last_checked = self.rate_limit
+            logger.debug("Remaining API points: {}, choosen index: {}".format(rls, max_idx))
+
+        # If we have any tokens use best of them
+        if len(self.token) > 0:
+            self.current_token = self.token[max_idx]
+            self.session.headers.update({'Authorization': 'token ' + self.current_token})
+
+    def _need_check_tokens(self):
+        """We need to consider changing token when last rate limit failed or allow using 10% of current token"""
```

Comment:
Would you reword these docstrings? You can use several lines if you need it. Try not to use first person forms as `we`. I also guess the comment to the 10% should be below (where the code does it). If one day we change the ratio limits we will have to remember to change the description of the method (and people usually forget that).

Reply:
Will address that.